### PR TITLE
Metal multi-pass support

### PR DESCRIPTION
**Pull request by kvark**
Fixes vkQuake:
gfx-quake

PR checklist:
- [x] `make` succeeds (on *nix)
- [x] `make reftests` succeeds
- [x] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

~~TODO: need to fix Dolphin~~

### CODE REVIEW

**Comment by kvark** [comment]
Everything works now :tada: , and this PR should be ready to land.
A small perf regression is expected because of disabled RP caching, but this is the only path forward now.

File: src/backend/metal/src/device.rs [hunk 1]
```diff
@@ -1128,7 +1189,7 @@ impl hal::Device<Backend> for Device {
             None => {
                 // TODO: This is a workaround for what appears to be a Metal validation bug
                 // A pixel format is required even though no attachments are provided
-                if pass_descriptor.main_pass.attachments.is_empty() {
+                if subpass.colors.is_empty() {
```
Comment (grovesNL) [inline src/backend/metal/src/device.rs]: Should we check `depth_stencil` here too?

File: src/backend/metal/src/device.rs [hunk 2]
```diff
@@ -1366,25 +1414,15 @@ impl hal::Device<Backend> for Device {
...
-        descriptor.set_render_target_array_length(extent.depth as NSUInteger);
-
-        let mut inner = n::FramebufferInner {
-            extent,
-            aspects: format::Aspects::empty(),
-            colors: SmallVec::new(),
-            depth_stencil: None,
-        };
-
+        /*
```
Comment (grovesNL) [inline src/backend/metal/src/device.rs]: Should this block just be removed entirely, or are we planning to revisit it soon?

File: src/backend/metal/src/device.rs [hunk 3]
```diff
@@ -860,10 +859,69 @@ impl hal::Device<Backend> for Device {
...
+                                self.private_caps.map_format(
+                                    attachments[id].format.expect("No depth-stencil format provided")
+                                ).expect("Unable to map depth-stencil format!")
+                            }),
+                    },
+                }
+            })
+            .collect();
+
+        // sprinkle load operations
```
Comment (grovesNL) [inline src/backend/metal/src/device.rs]: This makes sense, but might be useful to expand to the comments here a bit just to explain the intent. Also where do we distinguish between `loadOp` and `stencilLoadOp` (and store) for mixed depth stencil attachments?

**Comment by kvark** [comment]
Thanks for the review! All notes are addressed now.
Bors r=grovesNL

**Pull request merged**